### PR TITLE
SdkType: Add support for Linux and Windows

### PR DESCRIPTION
**Pull request by Turbo87**
as suggested in #91

### CODE REVIEW

File: src/org/rust/lang/RustSdkType.kt [hunk 1]
```diff
@@ -24,15 +24,32 @@ class RustSdkType : SdkType("Rust SDK") {
...
 
     override fun suggestHomePath(): String? {
-        if (SystemInfo.isMac) {
+        if (SystemInfo.isMac || SystemInfo.isLinux) {
             val multiRust = File(FileUtil.expandUserHome("~/.multirust/toolchains"))
             if (multiRust.exists()) {
                 return multiRust.absolutePath
             }
+
             val homebrew = File("/usr/local/Cellar/rust")
```
Comment (matklad) [inline src/org/rust/lang/RustSdkType.kt]: This is Mac specific, so probably the condition should be changed in some way.

File: src/org/rust/lang/RustSdkType.kt [hunk 2]
```diff
@@ -24,15 +24,32 @@ class RustSdkType : SdkType("Rust SDK") {
     override fun getIconForAddAction() = icon
 
     override fun suggestHomePath(): String? {
-        if (SystemInfo.isMac) {
+        if (SystemInfo.isMac || SystemInfo.isLinux) {
             val multiRust = File(FileUtil.expandUserHome("~/.multirust/toolchains"))
```
Comment (matklad) [inline src/org/rust/lang/RustSdkType.kt]: May be move this case to `if (SystemInfo.isUnix)`?

**Comment by matklad** [comment]
I wonder if there is something analogous to Python's [which](https://docs.python.org/3/library/shutil.html#shutil.which) in Java?

Rust adds itself to PATH on installation, so checking in PATH should be a more robust and cross platform solution

**Comment by Turbo87** [comment]
that won't work for multirust though...

File: src/org/rust/lang/RustSdkType.kt (hunk 2 above)
Comment (Turbo87) [inline src/org/rust/lang/RustSdkType.kt]: done

File: src/org/rust/lang/RustSdkType.kt (hunk 1 above)
Comment (Turbo87) [inline src/org/rust/lang/RustSdkType.kt]: done

**Comment by matklad** [comment]
You can `which multirust`

**Comment by matklad** [comment]
Hm, should we look for `rustc` or for `cargo`? Will there be any differences?

**Comment by Turbo87** [comment]
@matklad we are currently requiring both and since cargo is part of rust at least since v1.0 that should be okay. I would suggest to merge the current approach for now. If someone later comes up with a better solution using `which` we can still change it if it works better.

**Comment by atsky** [comment]
@matklad, `which multirust` will not help, we looking not for multirust exectable, but for `rustc` and `cargo` installed by multirust.

**Comment by matklad** [comment]
Hm, for multirust there is ` multirust which rustc`. This `which` is a sub command of multirust, not the hypothetical Java method.

**Comment by Turbo87** [comment]
but `multirust which rustc` will most likely also depend on the directory from which it is executed due to the `override` command... I fear that using `which` or something like that is actually more complicated than the current approach with only very small potential advantages.

**Comment by matklad** [comment]
Ok, I think we should merge the current solution. @atsky ?

**Comment by atsky** [comment]
@Turbo87, can you squash this commits. Three commtis for this change is too much.

**Comment by atsky** [comment]
And after that we can merge.

**Comment by Turbo87** [comment]
@atsky sure, done

**Pull request merged**